src/lsptoolshost/roslynLanguageServer.ts

```diff
@@ -662,6 +663,17 @@ function registerRazorCommands(context: vscode.ExtensionContext, languageServer:
     }));
 }
 
+function registerSourceGeneratedFilesContentProvider(context: vscode.ExtensionContext, languageServer: RoslynLanguageServer) {
+    context.subscriptions.push(vscode.workspace.registerTextDocumentContentProvider('source-generated', new class implements vscode.TextDocumentContentProvider {
+        // TODO: also implement eventing for when the file changes
```

Comment:
For now I'm just doing a small implementation of TextDocumentContentProvider here; the Omnisharp side has a fuller implementation that does support eventing but will require a bit of refactoring to be able to reuse.